plugins/templates/getTemplateDetails.js

```diff
@@ -0,0 +1,60 @@
+'use strict';
+
+const boom = require('@hapi/boom');
+const joi = require('joi');
+const schema = require('screwdriver-data-schema');
+const getSchema = schema.models.template.get;
+const idSchema = schema.models.template.extract('id');
+
+module.exports = () => ({
+    method: 'GET',
+    path: '/templates/{id}',
+    options: {
+        description: 'Get a single template',
+        notes: 'Returns a template record',
+        tags: ['api', 'templates'],
+        auth: {
+            strategies: ['token'],
+            scope: ['build', 'template', 'pipeline']
+        },
+
+        handler: async (request, h) => {
+            const { templateFactory  } = request.server.app;
+
+            return templateFactory
+                .get(request.params.id)
+                .then(async templateModel => {
+                    if (!templateModel) {
+                        throw boom.notFound('Template does not exist');
+                    }
+
+                    if (Array.isArray(templateFactory.environment)) {
+                        const data = await templateFactory.toJsonWithSteps();
+
+                        return h.response(data);
+                    }
+
+                    // convert environment obj to array
+                    const env = [];
+
+                    Object.keys(templateFactory.environment).forEach(name => {
+                        env.push({ [name]: templateFactory.environment[name] });
+                    });
+                    templateFactory.environment = env;
```

Comment:
2. All these chunks of code for `env` are not necessary. 

By the way, this code looks like a direct copy of https://github.com/screwdriver-cd/screwdriver/blob/31a9309855921adad629362ee8e5144e99674357/plugins/builds/get.js

Reply:
yes I took a ref of it.